File: PMS.BO/LoginRequest/LoginRequest.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PMS.BO
{
	public class LoginRequest
	{
		#region Property

		#region Email : String 
		public string Email { get; set; }
		#endregion

		#region UserName : String 
		public string UserName { get; set; }
		#endregion

		#region Password : String 
		public string Password { get; set; }
		#endregion

		#region ConfirmPassword : String 
		public string ConfirmPassword { get; set; }
		#endregion

		#region IsSSO : bool 
		public bool IsForSignIN { get; set; }
		#endregion

		#region IsSSO : bool 
		public bool IsSSO { get; set; }
		#endregion

		#region AuthorityType : EnumLoginType 
		public EnumLoginType AuthorityType { get; set; } = EnumLoginType.Normal;
		#endregion

		#endregion
	}
}

File: PMS/ClientApp/src/app/PMS_MODEL/Authentication/login-request.ts
import { EnumLoginType } from "../Enum";

export class LoginRequest {
    email: string = '';
    userName: string = '';
    password: string = '';
    confirmPassword: string = '';
    isForSignIN : boolean = true;
	isSSO : boolean = false;
    authorityType?: EnumLoginType = EnumLoginType.Normal;

    populateFromSSOForMicrosoft(account: any): void {
		this.email = account.username;
		this.userName = account.name;
		this.authorityType = EnumLoginType.Microsoft;
	}
	populateFromSSOForGoogle(account: any): void {
		this.email = account.email;
		this.userName = account.name;
		this.authorityType = EnumLoginType.Google;
	}
	populateFromSSOForNormal(account: any): void {
		this.email = account.email;
		this.userName = account.name;
		this.authorityType = EnumLoginType.Normal;
	}
}